### PR TITLE
[web] Polyfill emitters

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Using React Native internals will bundle `react-native` on web which can be avoided. 

# How

* Updated babel alias to use `react-native-web` emitters
* Updated `expo-react-native-adapter` to use polyfill for `RCTDeviceEventEmitter` on web
  * This should be removed in a future version of `react-native-web` pending **necolas/react-native-web#1275**
* Remove internal web polyfills from `expo`
  * This will stop bundling the extra web code **expo/expo-cli#409**
* Use latest `react-native-gesture-handler` with web polyfill
* Removed internal `react-native` reference from `expo-payments-stripe`

# Test Plan

NCL should run without throwing any error

### CODE REVIEW

File: packages/expo-react-native-adapter/src/EventEmitter.ts [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import invariant from 'invariant';
 import { Platform } from 'react-native';
-// Importing this directly will circumvent the webpack alias `react-native$`. This will enable us to
-// use NativeEventEmitter from React Native and not from RNWeb.
-import NativeEventEmitter from 'react-native/Libraries/EventEmitter/NativeEventEmitter';
+import NativeEventEmitter from './NativeEventEmitter';
+// import NativeEventEmitter from 'react-native/Libraries/EventEmitter/NativeEventEmitter';
```
Comment (ide) [inline packages/expo-react-native-adapter/src/EventEmitter.ts]: add a todo

File: packages/expo-react-native-adapter/src/RCTDeviceEventEmitter.web.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
```
Comment (ide) [inline packages/expo-react-native-adapter/src/RCTDeviceEventEmitter.web.ts]: rm

File: packages/expo-react-native-adapter/src/NativeEventEmitter.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Eventually just use react-native-web: https://github.com/necolas/react-native-web/pull/1275
```
Comment (ide) [inline packages/expo-react-native-adapter/src/NativeEventEmitter.ts]: Move these files to a src/vendor directory

**Review by ide — APPROVED** [review]

**Comment by brentvatne** [comment]
> Using React Native internals will bundle react-native on web which can be avoided.

so the purpose of this is that when libraries mistakenly import react-native internals, we alias that so that they don't end up bundling all of react-native in their app by accident? if so, this seems pretty reasonable as a way to avoid people shooting themselves in the foot. we need to also make some changes to react-native upstream to warn people when they import from internals to stop libraries from doing this so we can eventually get rid of this code

**Comment by EvanBacon** [comment]
@brentvatne yes just for the emitters though. We are going to be removing the safe-guards in favor of errors 🙃

**Pull request merged**